src/enum_set.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <bitset>
+
+template<typename T>
+class EnumSet {
+private:
+	std::bitset<8 * sizeof(typename std::underlying_type<T>::type)> bitmask {0};
+	int index_of(T t) const;
```

Comment:
`index_of` should be `static`. BTW you'll probably find it a lot more pleasant to define all these methods inline:

    static int index_of(T t) {
        static_cast<typename std::underlying_type<T>::type>(t);
    }

No more duplicated templates, etc.